src/Helpers.dfy

```diff
@@ -5,296 +5,12 @@
-      assert forall a, b :: IsMinimum(a, s, less) && IsMinimum(b, s, less) ==> a == b by {
-        // The proof of the assertion is the following forall statement.
-        // But why did we even bother to write the assert-by instead of
-        // just writing this forall statement in the first place? Because
-        // we are in an expression context and a forall statement cannot start
-        // an expression (because the "forall" makes the parser think that
-        // a forall expression is coming).
-        forall a, b | IsMinimum(a, s, less) && IsMinimum(b, s, less) {
-          // For the given a and b, the proof is settled by calling the following lemma:
-          MinimumIsUnique(a, b, s, less);
-        }
-      }
-      // The "a in s" in the following line follows from IsMinimum(a, s), so it
-      // is logically redundant. However, it is needed to convince the compiler
-      // that the assign-such-that statement is compilable.
-      var a :| a in s && IsMinimum(a, s, less);
-      [a] + SetToOrderedSequence(s - {a}, less)
-  }
-
-  predicate method IsMinimum<T(==)>(a: seq<T>, s: set<seq<T>>, less: (T, T) -> bool) {
-    a in s &&
-    forall z :: z in s ==> LexicographicLessOrEqual(a, z, less)
-  }
-
-  lemma ThereIsAMinimum<T>(s: set<seq<T>>, less: (T, T) -> bool)
-    requires s != {}
-    requires Trichotomous(less) && Transitive(less)
-    ensures exists a :: IsMinimum(a, s, less)
-  {
-    var a := FindMinimum(s, less);
-  }
-
-  lemma MinimumIsUnique<T>(a: seq<T>, b: seq<T>, s: set<seq<T>>, less: (T, T) -> bool)
-    requires IsMinimum(a, s, less) && IsMinimum(b, s, less)
-    requires Trichotomous(less)
-    ensures a == b
-  {
-    LexIsAntisymmetric(a, b, less);
-  }
+  export 
+    provides Comparison
 
-  lemma FindMinimum<T>(s: set<seq<T>>, less: (T, T) -> bool) returns (a: seq<T>)
-    requires s != {}
-    requires Trichotomous(less) && Transitive(less)
-    ensures IsMinimum(a, s, less)
-  {
-    a :| a in s;
-    if s == {a} {
-      LexIsReflexive(a, less);
-    } else {
-      var s' := s - {a};
-      assert forall x :: x in s <==> x == a || x in s';
-      var a' := FindMinimum(s', less);
-      if LexicographicLessOrEqual(a', a, less) {
-        a := a';
-      } else {
-        assert LexicographicLessOrEqual(a, a', less) by {
-          LexIsTotal(a, a', less);
-        }
-        forall z | z in s
-          ensures LexicographicLessOrEqual(a, z, less)
-        {
-          if z == a {
-            LexIsReflexive(a, less);
-          } else {
-            calc {
-              true;
-            ==  // z in s && z != a
-              z in s';
-            ==>  // by postcondition of FindMinim(s') above
-              LexicographicLessOrEqual(a', z, less);
-            ==>  { LexIsTransitive(a, a', z, less); }
-              LexicographicLessOrEqual(a, z, less);
-            }
-          }
-        }
-      }
-    }
-  }
 }
 
-module Comparison {
+module Helpers.Comparison {
```

Comment:
Can you please move this to Relations or somewhere else? That way you can delete this  now insignificant and mostly empty file